docs/modules/ROOT/pages/accounts.adoc

```diff
@@ -126,11 +128,31 @@ Where:
 NOTE: The scheme of building multicall transactions within the `\\__execute__` method will change once StarkNet allows for pointers in struct arrays.
 In which case, multiple transactions can be passed to (as opposed to built within) `\\__execute__`.
 
+There's a fourth canonical entrypoint for accounts, the `\\__validate_deploy__` method. It is **only callable by the protocol** during the execution of a `DeployAccount` type of transaction, but not by any other contract. This entrypoint is for counterfactual deployments.
+
+=== Counterfactual Deployments
+
+Counterfactual means something that hasn't happened.
+
+A deployment is said to be counterfactual when the deployed contract pays for it. It's called like this because we need to send the funds to the address before deployment. A deployment that hasn't happened.
+
+The steps are the following:
+
+1. Precompute the `address` given a `class_hash`, `salt`, and constructor `calldata`.
+2. Send funds to `address`.
+3. Send a `DeployAccount` type transaction.
+4. The protocol will then validate with `\\__validate_deploy__`.
+5. If successful, the protocol deploys the contract and the contract itself pays for the transaction.
+
+Since `address` will utimately depend on the `class_hash` and `calldata`, it's safe for the protocol to validate the signature and spend the funds on that address.
```

Comment:
```suggestion
Since `address` will ultimately depend on the `class_hash` and `calldata`, it's safe for the protocol to validate the signature and spend the funds on that address.
```